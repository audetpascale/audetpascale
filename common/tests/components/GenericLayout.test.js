/**
 * @jest-environment jsdom
 */
import React from "react";
import renderer from "react-test-renderer";
import { GenericLayout } from "../../index";
import { LocationProvider } from "@reach/router";

test("Generate Layout", () => {
  const component = renderer.create(
    <LocationProvider>
      <GenericLayout
        defaultSiteMetadata={{
          site: {
            siteMetadata: {
              author: "John Doe",
              description: { twitterUsername: "@johndoe" },
            },
          },
        }}
        menuItems={[]}
        footerImage={<img alt="footer logo" />}
        headerImage={<img alt="header logo" />}
        title="Jest"
        description="Hello world"
        image="Logo here"
        year={1900}
      >
        <span>Lorem ipsum</span>
      </GenericLayout>
    </LocationProvider>
  );
  let tree = component.toJSON();
  expect(tree).toMatchSnapshot();
});
