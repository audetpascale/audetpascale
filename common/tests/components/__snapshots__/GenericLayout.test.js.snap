// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Generate Layout 1`] = `
<div
  className="css-5y504r-GenericLayout"
>
  <header
    className="css-1upxki1-Header"
  >
    <nav
      className="css-13lif9l-Header"
    >
      <a
        className="css-yswi07"
        href="/"
      >
        <img
          alt="header logo"
        />
         Accueil
      </a>
      <div
        className="css-1eya205-Header"
      />
    </nav>
  </header>
  <main
    className="css-1t248nq-GenericLayout"
  >
    <span>
      Lorem ipsum
    </span>
  </main>
  <footer
    className="css-zm957j-Footer"
  >
    <span
      className="css-vurnku"
    >
      © 
      1900
       - 
      2021
       
      John Doe
      .
    </span>
    <div
      className="css-j4ym7v-Footer"
    />
    <img
      alt="footer logo"
    />
    <span
      className="css-vurnku"
    >
       est propulsé par 
      <a
        href="https://www.gatsbyjs.com/"
      >
        Gatsby
      </a>
       
      et 
      <a
        href="https://www.netlify.com/"
      >
        Netlify
      </a>
      .
    </span>
  </footer>
</div>
`;
